src/Lib.P2P.Tests/SwarmTest.cs

```diff
@@ -0,0 +1,1244 @@
+                "CAASXjBcMA0GCSqGSIb3DQEBAQUAA0sAMEgCQQDlTSgVLprWaXfmxDr92DJE1FP0wOexhulPqXSTsNh5ot6j+UiuMgwb0shSPKzLx9AuTolCGhnwpTBYHVhFoBErAgMBAAE="
+        };
+
+        [TestMethod]
+        public async Task Start_Stop()
+        {
+            var swarm = new SwarmService
+            {
+                LocalPeer = self
+            };
+            await swarm.StartAsync();
+            await swarm.StopAsync();
+        }
+
+        [TestMethod]
+        public void Start_NoLocalPeer()
+        {
+            var swarm = new SwarmService();
+            ExceptionAssert.Throws<NotSupportedException>(() =>
+            {
+                swarm.StartAsync().Wait();
+            });
+        }
+
+        [TestMethod]
+        public void NewPeerAddress()
+        {
+            var swarm = new SwarmService {LocalPeer = self};
+            swarm.RegisterPeerAddress(mars);
+            Assert.IsTrue(swarm.KnownPeerAddresses.Contains(mars));
+        }
+
+        [TestMethod]
+        public void NewPeerAddress_Self()
+        {
+            var swarm = new SwarmService {LocalPeer = self};
+            var selfAddress = "/ip4/178.62.158.247/tcp/4001/ipfs/" + self.Id;
+            ExceptionAssert.Throws<Exception>(() =>
+            {
+                var _ = swarm.RegisterPeerAddress(selfAddress);
+            });
+
+            selfAddress = "/ip4/178.62.158.247/tcp/4001/p2p/" + self.Id;
+            ExceptionAssert.Throws<Exception>(() =>
+            {
+                var _ = swarm.RegisterPeerAddress(selfAddress);
+            });
+        }
+
+        [TestMethod]
+        public void NewPeerAddress_BlackList()
+        {
+            var swarm = new SwarmService {LocalPeer = self};
+            swarm.BlackList.Add(mars);
+
+            ExceptionAssert.Throws<Exception>(() =>
+            {
+                var _ = swarm.RegisterPeerAddress(mars);
+            });
+            Assert.IsFalse(swarm.KnownPeerAddresses.Contains(mars));
+
+            Assert.IsNotNull(swarm.RegisterPeerAddress(venus));
+            Assert.IsTrue(swarm.KnownPeerAddresses.Contains(venus));
+        }
+
+        [TestMethod]
+        public void NewPeerAddress_WhiteList()
+        {
+            var swarm = new SwarmService {LocalPeer = self};
+            swarm.WhiteList.Add(venus);
+
+            ExceptionAssert.Throws<Exception>(() =>
+            {
+                var _ = swarm.RegisterPeerAddress(mars);
+            });
+            Assert.IsFalse(swarm.KnownPeerAddresses.Contains(mars));
+
+            Assert.IsNotNull(swarm.RegisterPeerAddress(venus));
+            Assert.IsTrue(swarm.KnownPeerAddresses.Contains(venus));
+        }
+
+        [TestMethod]
+        public void NewPeerAddress_InvalidAddress_MissingPeerId()
+        {
+            var swarm = new SwarmService {LocalPeer = self};
+            ExceptionAssert.Throws<Exception>(() =>
+            {
+                var _ = swarm.RegisterPeerAddress("/ip4/10.1.10.10/tcp/29087");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable '_'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)